enhancements/baremetal/ztp-metal3.md

```diff
@@ -0,0 +1,320 @@
+to Ironic and baremetal-operator. Otherwise, a failure will be recorded.
+
+The `CustomDeploy` field can be set from the very beginning as well. Then BMO
+will conduct inspection, apply firmware settings (if any) and prepare for
+the installation immediately.
+
+### Minimal ISO
+
+ICC is currently not capable of building a *mimimal ISO*, i.e. a CoreOS ISO
+without the root filesystems. While not a hard blocker of this work, the
+minimal ISO support should be implemented because
+- it improves compatibility with BMCs that do not accept large ISOs,
+- it reduces the resource footprint on the Ironic side.
+
+#### Notes
+
+- The new annotation with accept a URI instead of a full Ignition configuration
+  or a link to a secret because the configuration can be quite large when
+  many hosts are present.
+
+- ICC will replicate the graceful wait period logic from the assisted service:
+  the image will be built not earlier than 60 seconds from the last change
+  to the `PreprovisioningImage` object. This is done for two reasons: to give
+  the user enough time to apply all changes that may affect the build and to
+  reduce the probability of a race between BMAC and ICC.
+
+- Currently, BMO never deletes `PreprovisioningImage` resources. Since all
+  Ignition configurations are stored in memory, it may cause a significant
+  memory usage in the ZTP case. We will need to fix BMO to delete
+  `PreprovisioningImage` resources when the `BareMetalHost` becomes
+  `provisioned` or is deleted.
+
+- The same process will be applied to `initrd` images for (i)PXE booting.
+
+### Risks and Mitigations
+
+- More containers will be downloaded on the ramdisk start-up. We'll need to
+  account for more RAM and a longer start-up time.
+
+- Users will be responsible for correctly linking `BareMetalHost`s with
+  `InfraEnv` resources.
+
+## Design Details
+
+### Open Questions
+
+- The assisted mode of operation is to boot the assisted agent and wait for
+  instructions from the service (and indirectly - from the user). During this
+  time, the `BareMetalHost` state will be `provisioning`, and timeouts may
+  apply. We may need a way to tell Ironic to never time out this operation.
+
+### Test Plan
+
+- Building the new ISO will be covered by the existing ZTP e2e tests.
+- A new ZTP job will need to be created for network booting.
+
+### Graduation Criteria
+
+This work will be GA immediately since there is no phasing possible or planned.
+Only ZTP will be affected. The new behavior will be fully opt-in from the
+assisted service's (more specifically, BMAC) point of view.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+This change does not introduce new resources, so upgrading or downgrading them
+is not a concern.
+
+For the new flow to apply, it must be explicitly requested through the
+`BareMetalHost` API. Otherwise, the old flow is used.
```

Comment:
How long will the old workflow be kept in place?

Reply:
Sorry, I've complete misread your question....

There are no immediate plans to deprecate the live ISO workflow. I think I mention it somewhere else.